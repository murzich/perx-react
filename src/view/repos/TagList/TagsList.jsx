import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Label } from 'semantic-ui-react';
import styles from '../ReposItem/ReposItem.module.css';

const TagsList = ({ tags = {} }) => {
  const tagsArray = [
    {
      id: 'stars',
      default: 0,
      icon: 'star',
      value: tags.stars,
    },
    {
      id: 'forks',
      default: 0,
      icon: 'fork',
      value: tags.forks,
    },
    {
      id: 'watchers',
      default: 0,
      icon: 'eye',
      value: tags.watchers,
    },
    {
      id: 'language',
      default: null,
      value: tags.language,
    },
  ];

  return (
    <Label.Group
      size="small"
      className={styles.tags}
    >
      {tagsArray.map((tag) => {
        if (!tag.value && tag.default === null) {
          return null;
        }

        return (
          <Label key={tag.id}>
            {tag.icon && <Icon name={tag.icon} />}
            {tag.value || tag.default}
          </Label>
        );
      })}
    </Label.Group>
  );
};

TagsList.propTypes = {
  tags: PropTypes.shape({
    [PropTypes.string]: PropTypes.oneOf([PropTypes.string, PropTypes.number]),
  }),
};

export default TagsList;
