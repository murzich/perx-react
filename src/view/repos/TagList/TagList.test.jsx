import { cleanup, render } from '@testing-library/react';
import React from 'react';
import TagsList from './TagsList';

describe('TagList', () => {
  afterEach(cleanup);

  it('should match snapshot without props', () => {
    const { asFragment } = render(<TagsList />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with props', () => {
    const tags = {
      stars: 13,
      watchers: 42,
      forks: 1,
      language: 'test',
    };
    const { asFragment } = render(<TagsList tags={tags} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
