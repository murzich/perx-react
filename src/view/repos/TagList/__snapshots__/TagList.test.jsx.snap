// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList should match snapshot with props 1`] = `
<DocumentFragment>
  <div
    class="ui small labels tags"
  >
    <div
      class="ui label"
    >
      <i
        aria-hidden="true"
        class="star icon"
      />
      13
    </div>
    <div
      class="ui label"
    >
      <i
        aria-hidden="true"
        class="fork icon"
      />
      1
    </div>
    <div
      class="ui label"
    >
      <i
        aria-hidden="true"
        class="eye icon"
      />
      42
    </div>
    <div
      class="ui label"
    >
      test
    </div>
  </div>
</DocumentFragment>
`;

exports[`TagList should match snapshot without props 1`] = `
<DocumentFragment>
  <div
    class="ui small labels tags"
  >
    <div
      class="ui label"
    >
      <i
        aria-hidden="true"
        class="star icon"
      />
      0
    </div>
    <div
      class="ui label"
    >
      <i
        aria-hidden="true"
        class="fork icon"
      />
      0
    </div>
    <div
      class="ui label"
    >
      <i
        aria-hidden="true"
        class="eye icon"
      />
      0
    </div>
  </div>
</DocumentFragment>
`;
